src/Microsoft.AspNetCore.Razor.Evolution/BoundAttributeDescriptorComparer.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Internal;
+
+namespace Microsoft.AspNetCore.Razor.Evolution
+{
+    internal class BoundAttributeDescriptorComparer : IEqualityComparer<BoundAttributeDescriptor>
+    {
+        public static readonly BoundAttributeDescriptorComparer Default = new BoundAttributeDescriptorComparer();
+
+        protected BoundAttributeDescriptorComparer()
+        {
+        }
+
+        public virtual bool Equals(BoundAttributeDescriptor descriptorX, BoundAttributeDescriptor descriptorY)
+        {
+            if (descriptorX == descriptorY)
+            {
+                return true;
+            }
+
```

Comment:
I usually like to do `if (object.ReferenceEquals(x, y)) { return true; } ... if (x == null ^ y == null) { return false; }`